### PR TITLE
catalog-backend: throw correct error for not allowed entities

### PR DESCRIPTION
**Pull request by freben**

### CODE REVIEW

**Comment by freben** [comment]
(Side note: Wouldn't have been crazy to call this one `ForbiddenError` instead ... even though it's "leaking" its HTTP relationship)

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4333-z2vdgheokyfzo6pgy4wcpjn8fwp71gzt.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6017c79198552d8c45538ae2

File: .changeset/poor-ligers-flow.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-backend': patch
+---
+
+Throw `NotAllowedError` when registering locations with entities of disallowed kinds
```
Comment (OrkoHunter) [inline .changeset/poor-ligers-flow.md:5]: Vale should start using Urban Dictionary for unallowing "Unallowed". :D

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4333-z2vdgheokyfzo6pgy4wcpjn8fwp71gzt.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6017c79198552d8c45538ae2

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**